src/System.Management.Automation/engine/remoting/fanin/PriorityCollection.cs

```diff
@@ -154,15 +156,33 @@ internal void Add<T>(RemoteDataObject<T> data)
         /// </summary>
         internal void Clear()
         {
-            Dbg.Assert(null != _dataToBeSent, "Serialized streams are not initialized");
-            lock (_syncObjects[(int)DataPriorityType.PromptResponse])
+            /*
+                NOTE: Error paths during initialization can cause _dataSyncObjects to be null
+                causing an unhandled exception in finalize and a process crash.
+                Verify arrays and dataToBeSent objects before referencing.
+            */
+            if (null != _dataSyncObjects && null != _dataToBeSent)
             {
-                _dataToBeSent[(int)DataPriorityType.PromptResponse].Dispose();
-            }
+                int promptResponseIndex = (int)DataPriorityType.PromptResponse;
+                int defaultIndex = (int)DataPriorityType.Default;
 
-            lock (_syncObjects[(int)DataPriorityType.Default])
-            {
-                _dataToBeSent[(int)DataPriorityType.Default].Dispose();
+                lock (_dataSyncObjects[promptResponseIndex])
```

Comment:
Sorry for the late review.  I am not familiar with the error path but if _dataToBeSent array can contain null entries then it seems like so can _dataSyncObjects (_dataSyncObjects[promptResponseIndex] == null ?).

It seems like these arrays should be created in the PrioritySendDataCollection constructor...

Reply:
@PaulHigin: _dataSyncObjects is populated when it is created.  The entries are not null if the array is not null.
As far as when it is created, it appears the intent was to create both arrays at the same point.  I saw no strong reason to change it.

Reply:
But the _dataSyncObjects array is filled within a loop:
```
for (int i = 0; i < names.Length; i++)
    {
        _dataToBeSent[i] = new SerializedDataStream(_fragmentor.FragmentSize);
        _dataSyncObjects[i] = new object();
    }
```
If an exception is thrown within the loop then it is possible an array element is null.  Is this not the error path we are worried about?

Reply:
SerializedDataStream's ctor does not currently throw.

Note that it should (it relies on Dbg.Assert) and I opened an issue for this: https://github.com/PowerShell/PowerShell/issues/5800